src/effects.ts

```diff
@@ -12,16 +12,82 @@ export function action<Action>(action: Action): Effect<Action> {
   return [dispatch => dispatch(action)];
 }
 
-export function fromPromise<Action>(
-  promise: () => Promise<Action>
+export function fromPromise<Action, Value = unknown, Err = Error>(
+  promise: () => Promise<Value>,
+  ofSuccess: (value: Value) => Action,
+  ofError: (error: Err) => Action
+): Effect<Action>
+
+export function fromPromise<Action, Value = unknown, Err = Error>(
+  promise: () => Promise<Value>,
+  ofError: (error: Err) => Action
+): Effect<Action>
+
+export function fromPromise<Action, Value, Err>(
+  promise: () => Promise<Value>,
+  ofSuccess: ((value: Value) => Action) | ((error: Err) => Action),
+  ofError?: (error: Err) => Action
 ): Effect<Action> {
-  return [dispatch => promise().then(dispatch)];
+  const handleError = ofError === undefined
+    ? ofSuccess as ((error: Err) => Action)
+    : ofError
+
+  const handleSuccess = ofError === undefined
+    ? undefined
+    : ofSuccess as ((value: Value) => Action)
+
+  return [
+    (dispatch: Dispatch<Action>) => (
+      promise()
+        .then(value => {
+          if (handleSuccess !== undefined) {
+            dispatch(handleSuccess(value))
+          }
```

Comment:
So if handleSucess is undefined, the result of the promise is ignored?

Reply:
yes